dirsrvtests/tests/suites/memberof_plugin/regression_test.py

```diff
@@ -653,31 +687,39 @@ def test_entrycache_on_modrdn_failure(topology_st):
     _disable_auto_oc_memberof(topology_st.standalone)
     topology_st.standalone.restart(timeout=10)
 
-    # move group2 into the scope and check it fails
-    try:
-        topology_st.standalone.rename_s(group2_dn, 'cn=group_in2', newsuperior=peoplebase, delold=0)
-        topology_st.standalone.log.info("This is unexpected, modrdn should fail as the member entry have not the appropriate objectclass")
-        assert False
-    except ldap.OBJECT_CLASS_VIOLATION:
-        pass
-
-    # retrieve the entry having the specific description value
-    # check that the entry DN is the original group2 DN
-    ents = topology_st.standalone.search_s(DEFAULT_SUFFIX, ldap.SCOPE_SUBTREE, '(cn=gr*)')
-    found = False
-    for ent in ents:
-        topology_st.standalone.log.info("retrieve: %s with desc=%s" % (ent.dn, ent.getValue('description')))
-        if ent.getValue('description') == entry_description.encode():
-            found = True
-            assert ent.dn == group2_dn
-    assert found
+    if (memberof.get_memberofdeferredupdate() and memberof.get_memberofdeferredupdate().lower() == "on"):
+        # move group2 into the scope and check it fails
+        try:
+            topology_st.standalone.rename_s(group2_dn, 'cn=group_in2', newsuperior=peoplebase, delold=0)
+            topology_st.standalone.log.info("This is expected, modrdn does not fail only updates of members will fail")
+        except ldap.OBJECT_CLASS_VIOLATION:
+            pass
```

Comment:
This code looks odd to me...
- It doesn't have the `assert False` but it says `and check it fails`
- If it don't expect it to fail here, we probably should remove `try-except`